pkg/authconfig/authconfig_handler_test.go

```diff
@@ -0,0 +1,110 @@
+package authconfig_test
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/authconfig"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAuthClientConfigHandler(t *testing.T) {
+	// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
+	// pass 'nil' as the third parameter.
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/authconfig", nil)
+	require.NoError(t, err)
+
+	// Create logger and registry.
+	logger := log.New(os.Stderr, "", 0)
+	configRegistry := configuration.CreateEmptyRegistry()
+
+	// Set the config for testing mode, the handler may use this.
+	configRegistry.GetViperInstance().Set("testingmode", true)
+	assert.True(t, configRegistry.IsTestingMode(), "testing mode not set correctly to true")
+
+	// Create handler instance.
+	authconfigService := authconfig.New(logger, configRegistry)
+	handler := gin.HandlerFunc(authconfigService.AuthconfigHandler)
+
+	t.Run("valid json config", func(t *testing.T) {
+
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		rr := httptest.NewRecorder()
+		ctx, _ := gin.CreateTestContext(rr)
+		ctx.Request = req
+
+		handler(ctx)
+
+		// Check the status code is what we expect.
+		require.Equal(t, http.StatusOK, rr.Code)
+
+		// check response content-type.
+		require.Equal(t, configRegistry.GetAuthClientConfigAuthContentType(), rr.Header().Get("Content-Type"))
+
+		// Check the response body is what we expect.
+		// get config values from endpoint response
+		var data map[string]interface{}
+		err = json.Unmarshal(rr.Body.Bytes(), &data)
+		require.NoError(t, err)
+
+		// get the configured values
+		var config map[string]interface{}
```

Comment:
Ideally the configuration value tests should be in configuration tests. Not in the handler tests (because handlers doesn't care. It's just a raw string for the handler) and you already tested the strings in https://github.com/codeready-toolchain/registration-service/pull/17/files#diff-557f6e3c317a9e7443fff482bdcf1fc1R49

But I'm OK to keep it here :)